docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,195 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace"
+    }
+```
+Namespace updates can be set through adding a child in json form to the zookeeper path at`druid.zk.paths.namespacePath`. The json is of the following format:
+
+#### URI namespace update
+The remapping values for each namespace can be specified by a namespace update json object as per
+```json
+{
+  "namespace":{
+    "type":"uri",
+    "namespace":"some_namespace",
+    "uri": "file:///some/file/or/other/uri",
+    "parseSpec":{
+      "type":"csv",
+      "columns":["key","value]
+    }
+  },
+  "updateMs":0
```

Comment:
Yes, instead of having a "namespace" and "updateMS" the namespace spec will have updateMs in it for those which understand such a thing, and the "namespace" will be the top level object.